spec.go

```diff
@@ -123,8 +123,12 @@ func loadSpec(cPath string) (spec *specs.Spec, err error) {
 	}
 	defer cf.Close()
 
-	if err = json.NewDecoder(cf).Decode(&spec); err != nil {
-		return nil, err
+	if err = json.NewDecoder(cf).Decode(&spec); err != nil || spec == nil {
+		if err == nil {
+			return nil, fmt.Errorf("Invalid JSON specification file %s format", cPath)
+		} else {
+			return nil, err
+		}
 	}
```

Comment:
```suggestion
	if err = json.NewDecoder(cf).Decode(&spec); err != nil || spec == nil {
		return nil, err
	} else if spec == nil {
		return nil, fmt.Errorf("config cannot be null")
	}
```

It should be noted the issue here is not with invalid JSON (that would return an error from `json.Decode`), but instead with a `config.json` that is set to `null` (which is valid JSON). I feel like another option would be to have `spec` be `specs.Spec` which would stop it from being able to be `nil`.